src/BusTicket.java

```diff
@@ -1,13 +1,92 @@
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
 class BusTicket {
-    String ticketClass;
-    String type;
-    String startDate;
-    double price;
+    private final String ticketClass;
+    private final String ticketType;
+    private final String startDate;
+    private final double price;
 
-    public BusTicket(String ticketClass, String type, String startDate, double price) {
+    public BusTicket(String ticketClass, String ticketType, String startDate, double price) {
         this.ticketClass = ticketClass;
-        this.type = type;
+        this.ticketType = ticketType;
         this.startDate = startDate;
         this.price = price;
     }
+
+    public String getTicketClass() { return ticketClass; }
+    public String getTicketType() { return ticketType; }
+    public String getStartDate() { return startDate; }
+    public double getPrice() { return price; }
+
+    @Override
+    public String toString() {
+        return "BusTicket{" +
+                "ticketClass='" + ticketClass + '\'' +
+                ", ticketType='" + ticketType + '\'' +
+                ", startDate='" + startDate + '\'' +
+                ", price=" + price +
+                '}';
+    }
+
+    public static List<BusTicket> loadTicketsFromFile(String filePath) throws IOException {
```

Comment:
move this method to another class, reading from file not responsibility for model class. BusTicket should have only fields that related to object

Reply:
Also why method is static?

Reply:
the method is static because it’s function for reading tickets from the file and it doesn’t depend on any specific instance of the TicketFileReader class making it accessible directly without creating an object